python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -131,41 +133,6 @@ def setParams(self, inputCol=None, inputImageNodeName=None, outputCol=None,
         kwargs = self._input_kwargs
         return self._set(**kwargs)
 
-    def fit(self, dataset, params=None):
-        """
-        Fits a model to the input dataset with optional parameters.
-
-        .. warning:: This returns the byte serialized HDF5 file for each model to the driver.
-                     If the model file is large, the driver might go out-of-memory.
-                     As we cannot assume the existence of a sufficiently large (and writable)
-                     file system, users are advised to not train too many models in a single
-                     Spark job.
-
-        :param dataset: input dataset, which is an instance of :py:class:`pyspark.sql.DataFrame`.
-                        The column `inputCol` should be of type `sparkdl.image.imageIO.imgSchema`.
-        :param params: An optional param map that overrides embedded params. If a list/tuple of
-                       param maps is given, this calls fit on each param map and returns a list of
-                       models.
-        :return: fitted model(s). If params includes a list of param maps, the order of these
-                  models matches the order of the param maps.
-        """
-        self._validateParams()
-        if params is None:
-            paramMaps = [dict()]
-        elif isinstance(params, (list, tuple)):
-            if len(params) == 0:
-                paramMaps = [dict()]
-            else:
-                self._validateFitParams(params)
-                paramMaps = params
-        elif isinstance(params, dict):
-            self._validateFitParams(params)
```

Comment:
Remove `_validateFitParams` since it's currently unused? Also, do you know why we needed to check (in `_validateFitParams`) that the value of the input col param is not the name of one of the user-specified params? It seems like an unnecessary check to me.

Reply:
the logic `_validateFitParams` seems not useful and buggy, I've removed that.

@sueann @jkbradley @tomasatdatabricks  please note this change :)